### PR TITLE
[Atomic][X8664] set max atomic inline width according to the target

### PR DESCRIPTION
**Pull request by dcci**
This is to fix PR31620. MaxAtomicInlineWidth is set to 128 for x86_64. However
for target without cx16 support, 128 atomic operation will generate __sync_*
libcalls. The patch set MaxAtomicInlineWidth to 64 if the target doesn't support
cx16.

Differential Revision: https://reviews.llvm.org/D38046

git-svn-id: https://llvm.org/svn/llvm-project/cfe/trunk@313992 91177308-0d34-0410-b5e6-96231b3b80d8

### CODE REVIEW

**Comment by dcci** [comment]
@swift-ci please test

**Comment by dcci** [comment]
What's the way to force a test run? I don't have a Mac handy currently and I'm afraid this might break something.

**Comment by fredriss** [comment]
@swift-ci Please test

**Comment by fredriss** [comment]
@swift-ci please test

**Comment by dcci** [comment]
Allegedly this is failing because the python fetcher cannot checkout the correct branch?

```
Error on repo "/Users/buildnode/jenkins/workspace/swift-clang-PR-osx/swift": Traceback (most recent call last):
  File "/Users/buildnode/jenkins/workspace/swift-clang-PR-osx/swift/utils/update_checkout.py", line 126, in update_single_repository
    shell.run(['git', 'checkout', checkout_target], echo=True)
  File "/Users/buildnode/jenkins/workspace/swift-clang-PR-osx/swift/utils/swift_build_support/swift_build_support/shell.py", line 215, in run
    raise eout
Exception: ['git', 'checkout', u'swift-4.1-branch']
```

**Comment by fredriss** [comment]
yep, yep. This is the first clang PR on the 4.1 branch. It needs some tuning.

**Comment by dcci** [comment]
No worries, thanks. I thought I broke something instead :)

**Comment by fredriss** [comment]
Let's try this again
@swift-ci please test

**Comment by dcci** [comment]
The linux build is going, thanks!

**Comment by fredriss** [comment]
But the macOS one failed for a strange reason. @shahmishal can you take a look?

**Comment by shahmishal** [comment]
@swift-ci please test

**Comment by fredriss** [comment]
@dcci looks like you messed up the Target.cpp hunk

File: lib/Basic/Targets.cpp [hunk 1]
```diff
@@ -4938,6 +4938,14 @@ class X86_64TargetInfo : public X86TargetInfo {
     return llvm::makeArrayRef(BuiltinInfoX86,
                               X86::LastTSBuiltin - Builtin::FirstTSBuiltin);
   }
+
+  void setMaxAtomicWidth() override {
+    if (hasFeature("cx16"))
+      MaxAtomicInlineWidth = 128;
+    return;
+  }
+}
```
Comment (dcci) [inline lib/Basic/Targets.cpp]: sigh, this extra closing curly brace snuck in :(

**Comment by dcci** [comment]
@swift-ci please test

**Comment by dcci** [comment]
@swift-ci please test

**Comment by dcci** [comment]
It worked the second time, it just needed a kick ;)

**Comment by dcci** [comment]
It now passes. @fredriss can I go ahead and merge this one?

**Pull request merged**